src/core/recv_buffer.c

```diff
@@ -170,10 +182,74 @@ QuicRecvBufferIncreaseVirtualBufferLength(
     _In_ uint32_t NewLength
     )
 {
+    // TODO guhetier: Must make sure this can't be called in new mode
+    CXPLAT_DBG_ASSERT(RecvBuffer->RecvMode != QUIC_RECV_BUF_MODE_EXTERNAL);
     CXPLAT_DBG_ASSERT(NewLength >= RecvBuffer->VirtualBufferLength); // Don't support decrease.
     RecvBuffer->VirtualBufferLength = NewLength;
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+QUIC_STATUS
+QuicRecvBufferProvideChunks(
+    _Inout_ QUIC_RECV_BUFFER* RecvBuffer,
+    _Inout_ CXPLAT_LIST_ENTRY* /* QUIC_RECV_CHUNKS */ Chunks
+    )
+{
+    CXPLAT_DBG_ASSERT(!CxPlatListIsEmpty(Chunks));
+
+    //
+    // External chunks can be provided only if already in external mode, or if
+    // nothing has been written to the buffer yet.
+    //
+    uint64_t rangeMax;
```

Comment:
FYI, our style capitalizes all variables and fields.
```suggestion
    uint64_t RangeMax;
```
Please update for everything else too. Thanks!